### ISSUE TITLE
"Run clang-format on file save" should also format on Save All

### ISSUE THREAD

**Issue by AndiDog** [wangzw/CppStyle#16]
Ctrl+Shift+S shortcut (Save All) is often used by developers to save all modified files at once. In that case, CppStyle doesn't automatically format the content.

**Comment by arnetheduck** [comment]
I'm experiencing this too - it also happens when running Build and automatic Save All before build is enabled.

**Comment by wangzw** [comment]
Yes, it is because I do not know how to hook save all/automatic Save event... So any contribution is welcomed.

**Comment by Geobert** [comment]
I have absolutely no knowledge on Eclipse API but I found this, if it can inspire someone:
https://www.eclipse.org/forums/index.php/t/161995/

**Comment by shillingsburg** [comment]
I had some luck checking the "Format source code" box in Preferences->C/C++->Editor->Save Actions.

**Comment by Geobert** [comment]
> 
> 
> I had some luck checking the "Format source code" box in Preferences->C/C++->Editor->Save Actions.

This work only when you save *one* file, not when you click "save all"

**Comment by jmcarter17** [comment]
#32 Just realize that this is a doublon. Here is what I wrote on #32 : 

Hi, we have the same issue. We would like to see that "save all" or ctrl+shift+s (save all) formats the files also, if we have chosen to "Run clang-format on file save".

Also, if we have checked "Save automatically before build" in our workspace settings, it doesn't format on those saves either.

I believe all 3 cases (save-all button, ctrl-shift-s and save before build) are the same problem, but they should all, in my opinion, format the files. They are all "save all" operations.
